recipes-bsp/efibootguard/efibootguard_0.9.bb

```diff
@@ -52,6 +52,10 @@ addtask deploy before do_build after do_compile
 DEPENDS:class-native = "zlib-native libcheck-native"
 EXTRA_OECONF:class-native += "--disable-libpci"
 
+do_install:append () {
+    rm -f ${D}${libdir}/*.so
+}
```

Comment:
See https://github.com/siemens/efibootguard/commit/c8ad20f36145bb4cd5a61549fd96bb7cbde0f144 for a use case.